server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -24,20 +24,24 @@ import org.http4s.util.{CaseInsensitiveString, encodeHex}
   * When a user authenticates, `embedNew` is used to send a random CSRF value as a cookie.  (Alternatively,
   * an authenticating service can be wrapped in `withNewToken`).
   *
-  * For requests that are unsafe (PUT, POST, DELETE, PATCH), services protected by the `validated` method in the
+  * By default, for requests that are unsafe (PUT, POST, DELETE, PATCH), services protected by the `validated` method in the
   * middleware will check that the csrf token is present in both the header `headerName` and the cookie `cookieName`.
   * Due to the Same-Origin policy, an attacker will be unable to reproduce this value in a
   * custom header, resulting in a `401 Unauthorized` response.
   *
-  * Requests with safe methods (such as GET, OPTIONS, HEAD) will have a new token embedded in them if there isn't one,
+  * By default, requests with safe methods (such as GET, OPTIONS, HEAD) will have a new token embedded in them if there isn't one,
   * or will receive a refreshed token based off of the previous token to mitigate the BREACH vulnerability. If a request
   * contains an invalid token, regardless of whether it is a safe method, this middleware will fail it with
   * `401 Unauthorized`. In this situation, your user(s) should clear their cookies for your page, to receive a new
   * token.
   *
+  * The default can be overridden by modifying the `predicate` in `validate`. It will, by default, check if the method is safe.
+  * Thus, you can provide some whitelisting capability for certain kinds of requests.
+  *
   * We'd like to emphasize that you please follow proper design principles in creating endpoints, as to
   * not mutate in what should otherwise be idempotent methods (i.e no dropping your DB in a GET method, or altering
-  * user data). If you choose to not to, this middleware cannot protect you.
+  * user data). Please do not use the CSRF protection from this middleware as a safety net for bad design.
+  * Every time you mutate in a GET, a puppy dies. Think of the puppies. Don't be that guy.
```

Comment:
I chuckled, but in the interest of concision and not horrifying people unfamiliar with the idiom, this is probably a stronger scaladoc minus this line. :)